include/proxy-wasm/context.h

```diff
@@ -103,8 +143,17 @@ class ContextBase : public RootInterface,
   // The VM Context used for calling "malloc" has an id_ == 0.
   bool isVmContext() const { return id_ == 0; }
   // Root Contexts have the VM Context as a parent.
-  bool isRootContext() const { return root_context_id_ == 0; }
-  ContextBase *root_context() const { return root_context_; }
+  bool isRootContext() const { return parent_context_id_ == 0; }
+  ContextBase *parent_context() const { return parent_context_; }
+  ContextBase *root_context() const {
+    const ContextBase *previous = this;
+    ContextBase *parent = parent_context_;
+    while (parent != previous) {
+      previous = parent;
+      parent = parent->parent_context_;
```

Comment:
I think that will result in `nullptr` dereference when (accidentally) called on the root context itself. Maybe add a check?

Reply:
parent_context_ is never nullptr.  The list is terminated by parent_context_ == this.  This is enforced by the constructors for Context.